### PR TITLE
Add metrics for archival request counts

### PR DESCRIPTION
**Pull request by andrewjdawson2016**
These metrics enable us to produce availability graphs for archival.

### CODE REVIEW

File: service/worker/archiver/archiver.go [hunk 1]
```diff
@@ -131,6 +133,7 @@ func handleRequest(ctx workflow.Context, logger log.Logger, metricsClient metric
 			NonRetriableErrorReasons: deleteHistoryActivityNonRetryableErrors,
 		},
 	}
+	metricsClient.IncCounter(metrics.ArchiverScope, metrics.ArchiverDeleteWithRetriesCount)
```
Comment (shreyassrivatsan) [inline service/worker/archiver/archiver.go]: so effectively `ArchiverDeleteWithRetriesCount` will be equal to `ArchiverUploadWithRetriesCount` and be same as cadence requests..  I am ok with multiple metrics, but why do we need them?

File: service/worker/archiver/archiver.go [hunk 2]
```diff
@@ -113,6 +114,7 @@ func handleRequest(ctx workflow.Context, logger log.Logger, metricsClient metric
 		},
 	}
 	actCtx := workflow.WithActivityOptions(ctx, ao)
+	metricsClient.IncCounter(metrics.ArchiverScope, metrics.ArchiverUploadWithRetriesCount)
```
Comment (shreyassrivatsan) [inline service/worker/archiver/archiver.go]: technically you can use the `timertype:count` series for this and do not need a new separate emit here.. So then you will have the counts line up with the timer metrics exactly..

File: service/worker/archiver/archiver_test.go [hunk 3]
```diff
@@ -72,6 +75,7 @@ func (s *archiverSuite) TearDownTest() {
 }
 
 func (s *archiverSuite) TestHandleRequest_UploadFails_NonRetryableError() {
+
```
Comment (yycptt) [inline service/worker/archiver/archiver_test.go]: Empty line.

File: service/worker/archiver/archiver.go (hunk 2 above)
Comment (shreyassrivatsan) [reply service/worker/archiver/archiver.go]: something like this - fetch name:activity_info_count service:cadence-history deployment:prod-phx2  operation:sessionstats stats_type:count  timertype:count | transformNull 0 | aliasByTags name domain

Comment (andrewjdawson2016) [reply service/worker/archiver/archiver.go]: Oh interesting.

**Comment by andrewjdawson2016** [comment]
Using existing latency metrics instead of these counter metrics.

**Pull request closed without merging**